crates/dies-executor/src/roles/waller.rs

```diff
@@ -0,0 +1,157 @@
+use dies_core::BallData;
+use dies_core::PlayerData;
+use nalgebra::Vector2;
+
+use crate::{
+    roles::{skills::GoToPositionSkill, Role},
+    skill, PlayerControlInput,
+};
+
+
+pub struct Waller {
+    pub offset: f64,
+}
+use super::RoleCtx;
+
+impl Waller {
+    pub fn new(offset: f64) -> Self {
+        Self { offset }
+    }
+
+    fn find_intersection(&self, player_data: &PlayerData, ball: &BallData) -> Vector2<f64> {
+        let goal_center = Vector2::new(6117.0, -129.0); // Center of the goal area
+        let ball_pos = Vector2::new(ball.position.x, ball.position.y);
+
+        // Compute the direction vector from the ball to the goal center
+        let direction: Vector2<f64> = goal_center - ball_pos;
+        // Normalize the direction vector
+        let direction = direction.normalize();
+        println!("Direction: {:?}", direction);
```

Comment:
it's better to use the drawing functions instead of printing stuff
```suggestion
        dies_core::debug_line(ball_pos, ball_pos + direction, dies_core::DebugColor::Red);
```